In [6]:
# Song durations in seconds
song_durations = [210, 180, 245, 200, 175]

# Calculate total duration
total_duration = sum(song_durations)

print("Song durations:", song_durations)
print("Total playlist duration:", total_duration, "seconds")

Song durations: [210, 180, 245, 200, 175]
Total playlist duration: 1010 seconds


The list contains the duration of 5 songs in seconds.

Using the `sum()` function, we calculate the total duration of all songs.

Total duration = 210 + 180 + 245 + 200 + 175 = 1010 seconds.

Therefore, the total playlist duration is **1010 seconds**.

In [7]:
# 3x3 matrix of ratings
ratings = [
    [5, 4, 3],
    [4, 5, 2],
    [3, 4, 5]
]

# Print the matrix
print("Ratings Matrix:")
print(ratings)

# Rating given by second friend to second food item
rating = ratings[1][1]

print("Rating given by second friend to second food item:", rating)

Ratings Matrix:
[[5, 4, 3], [4, 5, 2], [3, 4, 5]]
Rating given by second friend to second food item: 5


The 3×3 matrix stores ratings given by 3 friends to 3 food items.

The rating given by the second friend to the second food item is **5**.

Python uses zero-based indexing, so we access this value using `ratings[1][1]`.

In [8]:
# Rows = days
# Columns = dishes

orders = [
    [10, 15, 20, 12],
    [12, 18, 15, 10],
    [8,  20, 22, 14]
]

# Calculate total orders for each dish
total_orders = [0, 0, 0, 0]

for row in orders:
    for i in range(4):
        total_orders[i] += row[i]

print("Total orders for each dish:", total_orders)

Total orders for each dish: [30, 53, 57, 36]


The matrix contains orders for 4 dishes across 3 days.

By adding the orders for each column, the total orders are:

- Dish 1: 30
- Dish 2: 53
- Dish 3: 57
- Dish 4: 36

Therefore, Dish 3 has the highest total number of orders.

In [9]:
# Matrix: [price, quantity]
products = [
    [500, 10],
    [800, 5],
    [1200, 4],
    [300, 15],
    [1500, 3]
]

# Function to calculate revenue
def calculate_revenue(products):
    revenues = []

    for row in products:
        price = row[0]
        quantity = row[1]
        
        revenue = price * quantity
        revenues.append(revenue)

    return revenues


# Calculate total revenue
revenues = calculate_revenue(products)

print("Product revenues:", revenues)

Product revenues: [5000, 4000, 4800, 4500, 4500]


The matrix stores the price and quantity sold for 5 products.

For each product, total revenue is calculated using:

Revenue = Price × Quantity

The revenues are:

- Product 1: ₹5,000
- Product 2: ₹4,000
- Product 3: ₹4,800
- Product 4: ₹4,500
- Product 5: ₹4,500

Therefore, Product 1 generated the highest revenue of ₹5,000.